pyne/mcnp.py

```diff
@@ -1399,13 +1399,24 @@ def mat_from_inp_line(filename, mat_line, densities='None'):
     # collect all material card data on one string
     line_index = 1
     line = linecache.getline(filename, mat_line + line_index)
+    lib_id = {}
+    lib_name = ['nlib', 'plib', 'hlib', 'pnlib', 'elib']
     # people sometimes put comments in materials and then this loop breaks                                                                                       # so we need to keep reading if we encounter comments
     while len(line.split()) > 0 and (line[0:5] == '     ' or line[0].lower() == 'c'):
+        
+        # This line reads deafult cross-section library from material card 
+        for token in line.split():
```

Comment:
Why not do this at about 1443 where you are processing `data_string`?  This is a code maintainability question related to separation of concerns.  The overarching purpose of this loop is to filter out information that doesn't need to be processed.  The next loop will goes through this filtered string to process the necessary info.  It makes more sense to process the default lib info there.

Reply:
Solved!